pkg/controller/servicebindingrequest/nested/nested.go

```diff
@@ -0,0 +1,181 @@
+)
+
+// getValueFromMap attempts to retrieve from `obj` a value from the given path
+// `path`.
+func getValueFromMap(obj map[string]interface{}, path Path) (interface{}, bool, error) {
+	head, exists := path.Head()
+	if !exists {
+		return obj, true, nil
+	}
+	val, exists := obj[head.Name]
+	if !exists {
+		return nil, false, nil
+	}
+
+	return getValue(val, path.Tail())
+}
+
+// collectValues accumulates the values found in all the given `path` in all
+// elements present in `obj`.
+func collectValues(obj []interface{}, path Path) (interface{}, error) {
+	r := accumulator.NewAccumulator()
+	for _, e := range obj {
+		val, found, err := getValue(e, path.AdjustedPath())
+		if err != nil || !found {
+			return nil, err
+		}
+		err = r.Accumulate(val)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return r.Value(), nil
+}
+
+// UnsupportedTypeErr is returned when an unsupported type is encountered.
+var UnsupportedTypeErr = errors.New("unsupported type")
+
+// convertToSlice attempts to convert the given `src` into a `[]interface{}`. An
+// error is returned when `src` is not one of the following:
+//
+// - []map[string]interface{}
+// - []string
+// - []int
+//
+func convertToSlice(src interface{}) ([]interface{}, error) {
+	var obj []interface{}
+	switch t := src.(type) {
+	case []map[string]interface{}:
+		obj = make([]interface{}, len(t))
+		for i, e := range t {
+			obj[i] = e
+		}
+	case []string:
+		obj = make([]interface{}, len(t))
+		for i, e := range t {
+			obj[i] = e
+		}
+	case []int:
+		obj = make([]interface{}, len(t))
+		for i, e := range t {
+			obj[i] = e
+		}
+	default:
+		return nil, UnsupportedTypeErr
+	}
+	return obj, nil
+}
+
+// InvalidIndexErr is returned when a given index is out of bounds.
+var InvalidIndexErr = errors.New("invalid index")
+
+// getValueFromSlice attempts to return the value present at the given path.
+func getValueFromSlice(s interface{}, path Path) (interface{}, bool, error) {
+	// assert and convert s to []interface{}
+	obj, err := convertToSlice(s)
+	if err != nil {
+		return nil, false, err
+	}
+
+	// it is required for path to have a head in this case, since it is expected to contain either an
+	// Index or a sub-key in order to extract or aggregate the underlying value.
+	head, ok := path.Head()
+	if !ok {
+		return nil, false, nil
+	}
+
+	if head.Index != nil {
+		if *head.Index > len(obj) {
+			return nil, false, InvalidIndexErr
+		}
+		m := obj[*head.Index]
+		return getValue(m, path.Tail())
+	}
+
+	r, err := collectValues(obj, path)
+	if err != nil {
+		return nil, false, err
+	}
+	return r, true, nil
+}
+
+// getValue attempts to return the value present at the given path.
+func getValue(obj interface{}, path Path) (interface{}, bool, error) {
+	// return obj if path is empty.
+	if _, ok := path.Head(); !ok {
+		return obj, true, nil
+	}
+
+	switch val := obj.(type) {
+	case string, int: // scalar
```

Comment:
In certain cases, we consider types like `int64` and `float64`.
Refer - https://github.com/redhat-developer/service-binding-operator/pull/407/files#diff-3e6f3c5d6b364a5c825abdf9b051d8adR64

Is there a reason for not considering those types here?